### ISSUE TITLE
about the task configuration

### ISSUE THREAD

**Issue by JesseBaomihua** [didi/delta#89]
what's the meaning of each option in the .yml file,Are there any articles devoted to this?

**Comment by zh794390558** [comment]
Sorry, we does not have an introduction on `yaml` config file. But you can see the code, or list which field that you do not know.

**Comment by JesseBaomihua** [comment]
Now I want to start a Slots Filling task,what's the key points for the configuration in the yml file?
like this

model:
  name: JointBilstmCrfModel
  type: keras # raw, keras or eager model
  use_pre_train_emb: False

or  Can you tell me which part of those options determine the choice of different algorithms?

**Comment by luffywalf** [comment]
1. Most parameters in the configuration works somewhere and looking at the code is the best way to know them. Slot filling task belongs to sequence label task,  whose parameters about data processing are mostly in init() function(delta/data/task/text_seq_label_task.py) and parameters about model are mostly in init() function (delta/models/text_seq_label_model.py).

2. here are some explanations for parameters which i think are important for a quick start
(it works using parameters default in configuration, but maybe you want try something else :P )
---
  task:
    name: TextSeqLabelTask  # sequence labeling task  name
    language: english    # choose language
    split_by_space: true  # if you want tokenize data by space, set true
    vocab_min_frequency: 0  # words whose frequency are lower than this number will be removed
    text_vocab: text vocab dicts generated could be found in this path
    label_vocab: label dict path
    max_seq_len: 150 #max length for  sequence length
    batch_size: 120
    epochs: 60
    shuffle_buffer_size: 15000 # equal or larger than data_size will be better for data shuffling
    need_shuffle: true
    classes:
      num_classes: 9 # number of classes for sequence labeling

model:
  name: BilstmCrfModel # model name that you choose to use
  use_pre_train_emb: false # if use pre_train_emb, such as word2vec...
  net:  # parameters for model 
    structure:
      embedding_size: 300
      cell_type: lstm
      cell_dim: 100
      num_units: 300
      num_layers: 1
      batch_size: 120
      dropout_rate: 0.5
      fc_dim: 100

solver: # parameters about training or eval.....
  name: RawSeqLabelSolver
    optimizer:
    name: gradientdecent 
    loss: CrfLoss 
    learning_rate:
      rate: 0.1 # learning rate of Adam optimizer
      type:  exp_decay # learning rate type
      decay_rate: 0.99  # the lr decay rate
      decay_steps: 100  # the lr decay_step for optimizer

hope it helps!

**Comment by hankun11** [comment]
> Sorry, we does not have an introduction on `yaml` config file. But you can see the code, or list which field that you do not know.

@zh794390558  It is our responsibility to provide the description for the options on `yaml`

**Comment by zh794390558** [comment]
@hankun11 ok, but I does not have time to this, please assign this to other.
